core/src/main/scala/chisel3/VerificationStatement.scala

```diff
@@ -66,7 +79,32 @@ object assert {
   ): c.Tree = {
     import c.universe._
     val apply_impl_do = symbolOf[this.type].asClass.module.info.member(TermName("_applyWithSourceLine"))
-    q"$apply_impl_do($cond, ${getLine(c)}, _root_.scala.Some($message), ..$data)($sourceInfo, $compileOptions)"
+    q"$apply_impl_do($cond, ${getLine(c)},_root_.scala.Some(_root_.chisel3.Printable.pack($message,..$data)))($sourceInfo, $compileOptions)"
+  }
+
+  def _applyMacroWithStringMessage(
```

Comment:
```suggestion
  private def _applyMacroWithStringMessage(
```
What happens if we make this private (or package private)? I'm not certain that we can but it's worth trying.

Reply:
Good point - let me give this a shot!

Reply:
@jackkoenig  - sadly this gives an error
`VerificationStatement.scala:44:21: macro implementation must be public
[error]   ): Assert = macro _applyMacroWithStringMessage`

Reply:
Let me try package private ....

Reply:
Same error - this is what I tried
`private[chisel3] def _applyMacroWithStringMessage(`

Reply:
So I guess we will have to live with this being visible - and just probably add a WARNING comment to not use this directly.

Reply:
Added a comment indicating not to use this directly and only for use within the macros.